### ISSUE TITLE
Podman logs output include blank lines occasionally

### ISSUE THREAD

**Issue by ypu** [containers/libpod#601]
/kind bug

**Description**

The output of podman logs sometimes include blank lines but sometimes not. This will influence the podman logs tests, and makes the ci test failed time to time.

**Steps to reproduce the issue:**

1. Run podman logs task

2. Check the logs file and test results

**Describe the results you received:**
Some output from our CI task:

Running: /usr/bin/podman --root /tmp/podman_test397002604/crio --runroot /tmp/podman_test397002604/crio-run --runtime /usr/bin/runc --conmon /usr/libexec/crio/conmon --cni-config-dir /etc/cni/net.d --storage-driver=vfs run -dt docker.io/library/alpine:latest sh -c echo podman; echo podman; echo podman
d2bb102e7aa2eca7d7056856a9ac2d9b215d1300d07db200aeb8c22272383bfa
Running: /usr/bin/podman --root /tmp/podman_test397002604/crio --runroot /tmp/podman_test397002604/crio-run --runtime /usr/bin/runc --conmon /usr/libexec/crio/conmon --cni-config-dir /etc/cni/net.d --storage-driver=vfs logs d2bb102e7aa2eca7d7056856a9ac2d9b215d1300d07db200aeb8c22272383bfa
podman

podman

podman
......

• Failure [5.262 seconds]
Podman logs
/go/src/github.com/projectatomic/libpod/test/e2e/logs_test.go:10
  podman logs for container [It]
  /go/src/github.com/projectatomic/libpod/test/e2e/logs_test.go:32

  Expected
      <int>: 7
  to equal
      <int>: 4

**Describe the results you expected:**

Output from logs should not include blank lines and looks like:
podman
podman
podman

**Comment by rhatdan** [comment]
@ypu Do we still have this bug?

**Comment by ypu** [comment]
Yes, test with the newest code it still happen in my fedora host. Here is the version infro of podman:

# podman version
Version:       0.6.2-dev
Go Version:    go1.10.2
OS/Arch:       linux/amd64

**Comment by rhatdan** [comment]
```
sh-4.4# podman rm test; podman run --name test -ti fedora sh -c "echo podman;echo podman; echo podman"; podman logs test
4f5b4ce99bd3803001373775ee54b0b96b93f3b07914bda484233835117e4977
podman
podman
podman
podman
podman
podman
sh-4.4# podman rm test; podman run --name test -ti fedora sh -c "echo podman;echo podman; echo podman"; podman logs test
e61e4b5c27079f5ff652b4bfcd2d946f3eb31d9b0f6e5408b88f6aa5c440abd6
podman
podman
podman
podman
podman
podman
sh-4.4# podman rm test; podman run --name test -ti fedora sh -c "echo podman;echo podman; echo podman"; podman logs test
c9d15e19530cc3eab83fc9c75fad0db367e124930476736ae6815b097e0df380
podman
podman
podman
podman
podman
podman
sh-4.4# podman rm test; podman run --name test -ti fedora sh -c "echo podman;echo podman; echo podman"; podman logs test
65739247ae5df02450c05b5571dcf5fb880b9f6e79645a393a72fb5bc5f32ae9
podman
podman
podman
podman
podman
podman
sh-4.4# podman rm test; podman run --name test -ti fedora sh -c "echo podman;echo podman; echo podman"; podman logs test
7648738c14a701512da35323ee418de29a18cf1f26d7d46e2c8982fb1969cf2c
podman
podman
podman
podman
podman
podman
```
I can't get it to happen?

**Comment by ypu** [comment]
Hi @rhatdan 

I test it with a newly installed f28 server and the output of podman logs still have problem. And following is the results: 

```
[root@dhcp-3-14 libpod]# podman rm test; podman run --name test -ti fedora sh -c "echo podman;echo podman; echo podman"; podman logs test
no container with name or ID test found: no such container
unable to find container test: no container with name or ID test found: no such container
Trying to pull docker.io/fedora:latest...Getting image source signatures
Copying blob sha256:e71c36a80ba912dd7a5a9f2f2d6136c148afa19bc7d024bd616b74a0bc7a2774
 82.57 MB / 82.57 MB [=====================================================] 54s
Copying config sha256:cc510acfcd701a409014118d5f417f0022520802a26c650866b8a9594d75f3a7
 2.29 KB / 2.29 KB [========================================================] 0s
Writing manifest to image destination
Storing signatures
podman
podman
podman
podman

podman

podman
```

**Comment by rhatdan** [comment]
Strange, I am not seeing this, could this be something to do with language?  redirect the output of podman logs test to a file and see if their are extra white spaces in the file or extra \n?

podman is reading the ctr.log file

```
# podman inspect --format '{{.LogPath}}' test
/var/lib/containers/storage/overlay-containers/df7d92f6488d63b89e2d1355680558cb1205817ba52860ef6d1fa48087f31c89/userdata/ctr.log
```
Could you examine that file to see if it has extra text in it.

**Comment by ypu** [comment]
Hi @rhatdan 
Here is the content of the log file:

```
[root@dhcp-3-14 ~]# podman inspect --format '{{.LogPath}}' test
/var/lib/containers/storage/overlay-containers/c965a47d88fd790fc897b53f3b3dfe18a969d4e760a2699a4238d7e4c7dd3ada/userdata/ctr.log
[root@dhcp-3-14 ~]# cat /var/lib/containers/storage/overlay-containers/c965a47d88fd790fc897b53f3b3dfe18a969d4e760a2699a4238d7e4c7dd3ada/userdata/ctr.log
2018-06-06T17:43:28.250628428+08:00 stdout P podman
2018-06-06T17:43:28.251291556+08:00 stdout F 
2018-06-06T17:43:28.251906050+08:00 stdout P podman
2018-06-06T17:43:28.252324975+08:00 stdout F 
2018-06-06T17:43:28.252835514+08:00 stdout P podman
2018-06-06T17:43:28.253189349+08:00 stdout F 
```

**Comment by rhatdan** [comment]
For the same tests I am seeing.
```
podman inspect --format '{{.LogPath}}' test
/var/lib/containers/storage/overlay-containers/c965a47d88fd790fc897b53f3b3dfe18a969d4e760a2699a4238d7e4c7dd3ada/userdata/ctr.log
cat /var/lib/containers/storage/overlay-containers/bf9e7e6ee5d2654e069943ad96f132e0aa6d010044c62ea50810c54729b398de/userdata/ctr.log
2018-06-07T09:12:07.677952896-04:00 stdout F podman
2018-06-07T09:12:07.677952896-04:00 stdout F podman
2018-06-07T09:12:07.677952896-04:00 stdout F podman
```
@mrunalp @mheon Any ideas?

**Comment by rhatdan** [comment]
@ypu What language setting are you using?
```
 echo $LANG
en_US.UTF-8
```
Could this be some unicode interation?

**Comment by ypu** [comment]
My $LANG in the test machine is here:

```
# echo $LANG
en_US.utf8
```

And the env is like this:

```
# env
LS_COLORS=rs=0:di=38;5;33:ln=38;5;51:mh=00:pi=40;38;5;11:so=38;5;13:do=38;5;5:bd=48;5;232;38;5;11:cd=48;5;232;38;5;3:or=48;5;232;38;5;9:mi=01;05;37;41:su=48;5;196;38;5;15:sg=48;5;11;38;5;16:ca=48;5;196;38;5;226:tw=48;5;10;38;5;16:ow=48;5;10;38;5;21:st=48;5;21;38;5;15:ex=38;5;40:*.tar=38;5;9:*.tgz=38;5;9:*.arc=38;5;9:*.arj=38;5;9:*.taz=38;5;9:*.lha=38;5;9:*.lz4=38;5;9:*.lzh=38;5;9:*.lzma=38;5;9:*.tlz=38;5;9:*.txz=38;5;9:*.tzo=38;5;9:*.t7z=38;5;9:*.zip=38;5;9:*.z=38;5;9:*.Z=38;5;9:*.dz=38;5;9:*.gz=38;5;9:*.lrz=38;5;9:*.lz=38;5;9:*.lzo=38;5;9:*.xz=38;5;9:*.zst=38;5;9:*.tzst=38;5;9:*.bz2=38;5;9:*.bz=38;5;9:*.tbz=38;5;9:*.tbz2=38;5;9:*.tz=38;5;9:*.deb=38;5;9:*.rpm=38;5;9:*.jar=38;5;9:*.war=38;5;9:*.ear=38;5;9:*.sar=38;5;9:*.rar=38;5;9:*.alz=38;5;9:*.ace=38;5;9:*.zoo=38;5;9:*.cpio=38;5;9:*.7z=38;5;9:*.rz=38;5;9:*.cab=38;5;9:*.wim=38;5;9:*.swm=38;5;9:*.dwm=38;5;9:*.esd=38;5;9:*.jpg=38;5;13:*.jpeg=38;5;13:*.mjpg=38;5;13:*.mjpeg=38;5;13:*.gif=38;5;13:*.bmp=38;5;13:*.pbm=38;5;13:*.pgm=38;5;13:*.ppm=38;5;13:*.tga=38;5;13:*.xbm=38;5;13:*.xpm=38;5;13:*.tif=38;5;13:*.tiff=38;5;13:*.png=38;5;13:*.svg=38;5;13:*.svgz=38;5;13:*.mng=38;5;13:*.pcx=38;5;13:*.mov=38;5;13:*.mpg=38;5;13:*.mpeg=38;5;13:*.m2v=38;5;13:*.mkv=38;5;13:*.webm=38;5;13:*.ogm=38;5;13:*.mp4=38;5;13:*.m4v=38;5;13:*.mp4v=38;5;13:*.vob=38;5;13:*.qt=38;5;13:*.nuv=38;5;13:*.wmv=38;5;13:*.asf=38;5;13:*.rm=38;5;13:*.rmvb=38;5;13:*.flc=38;5;13:*.avi=38;5;13:*.fli=38;5;13:*.flv=38;5;13:*.gl=38;5;13:*.dl=38;5;13:*.xcf=38;5;13:*.xwd=38;5;13:*.yuv=38;5;13:*.cgm=38;5;13:*.emf=38;5;13:*.ogv=38;5;13:*.ogx=38;5;13:*.aac=38;5;45:*.au=38;5;45:*.flac=38;5;45:*.m4a=38;5;45:*.mid=38;5;45:*.midi=38;5;45:*.mka=38;5;45:*.mp3=38;5;45:*.mpc=38;5;45:*.ogg=38;5;45:*.ra=38;5;45:*.wav=38;5;45:*.oga=38;5;45:*.opus=38;5;45:*.spx=38;5;45:*.xspf=38;5;45:
SSH_CONNECTION=10.66.14.247 36446 10.66.3.14 22
LANG=en_US.utf8
HISTCONTROL=ignoredups
HOSTNAME=dhcp-3-14.nay.redhat.com
XDG_SESSION_ID=1
USER=root
GOPATH=/root/go
SELINUX_ROLE_REQUESTED=
PWD=/root
HOME=/root
SSH_CLIENT=10.66.14.247 36446 22
SELINUX_LEVEL_REQUESTED=
SSH_TTY=/dev/pts/0
MAIL=/var/spool/mail/root
TERM=xterm-256color
SHELL=/bin/bash
XMODIFIERS=@im=none
SELINUX_USE_CURRENT_RANGE=
SHLVL=1
LOGNAME=root
DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/0/bus
XDG_RUNTIME_DIR=/run/user/0
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/root/go/bin:/root/bin
HISTSIZE=1000
LESSOPEN=||/usr/bin/lesspipe.sh %s
_=/usr/bin/env
```

**Comment by mheon** [comment]
Can't reproduce this over here as well. @ypu Are you testing on a VM? AWS?

**Comment by ypu** [comment]
Hi @mheon 
I use a kvm vm and installed it with fedora 28 release server. The system is updated after install.

**Comment by rhatdan** [comment]
@ypu Can you let me access this machine, via private email.

**Comment by rhatdan** [comment]
Urvashi could you take a look at this @mrunalp suggests that there is code in kubernetes that we can take that handles the processing of the log and properly handles the F and P Flags, still do not know why this machine works differently then our  machines.  But the way we handle this now is broken.

**Comment by umohnani8** [comment]
@rhatdan ACK. @ypu can I get access to the environment as well.

**Comment by rhatdan** [comment]
Actually you don't need access you should be able to just create his log file

**Comment by rhatdan** [comment]
```
# cat /var/lib/containers/storage/overlay-containers/c965a47d88fd790fc897b53f3b3dfe18a969d4e760a2699a4238d7e4c7dd3ada/userdata/ctr.log
2018-06-06T17:43:28.250628428+08:00 stdout P podman
2018-06-06T17:43:28.251291556+08:00 stdout F 
2018-06-06T17:43:28.251906050+08:00 stdout P podman
2018-06-06T17:43:28.252324975+08:00 stdout F 
2018-06-06T17:43:28.252835514+08:00 stdout P podman
2018-06-06T17:43:28.253189349+08:00 stdout F 
```

**Comment by rhatdan** [comment]
I hand edited mine.  This was happening on just one machine.

**Comment by vrothberg** [comment]
The K8s code can be found here: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kuberuntime/logs/logs.go#L128

Considering CRI-O is using more and more libpod code, I propose to add the logging functionality to `libpod` such that Podman and CRI-O (and any other libpod user) can take advantage of that.

@baude @mheon @umohnani8 @rhatdan WDYT?

**Comment by umohnani8** [comment]
@vrothberg thanks for the info! Will work on getting something out today.

**Comment by umohnani8** [comment]
Fixed in #1403
